package/MDAnalysis/coordinates/FHIAIMS.py

```diff
@@ -0,0 +1,322 @@
+
+where x,y, and z are expressed in Angstrom, and nx, ny and nz are
+real numbers in [0,1] and are used to compute the atomic positions
+in units of the basic cell.
+
+The field name is a string identifying the atomic species.
+See also the specifications in the official `FHIAIMS format`_.
+
+Classes
+-------
+
+.. autoclass:: Timestep
+   :members:
+
+.. autoclass:: FHIAIMSReader
+   :members:
+
+.. autoclass:: FHIAIMSWriter
+   :members:
+
+
+Developer notes: ``FHIAIMSWriter`` format strings
+-------------------------------------------------
+
+The :class:`FHIAIMSWriter` class has a :attr:`FHIAIMSWriter.fmt`
+attribute, which is a dictionary of different strings for writing
+lines in ``.in`` files.  These are as follows:
+
+``xyz``
+  An atom line without velocities.  Requires that the 'name' and
+  'pos' keys be supplied.  E.g.::
+
+     fmt['xyz'].format(pos=(0.0, 1.0, 2.0), name='O')
+
+``box_triclinic``
+  The (optional) initial lines of the file which gives box dimensions.
+  Requires the box keyword, as a length 9 vector.  This is a flattened
+  version of the (3,3) triclinic vector representation of the unit
+  cell.
+
+.. _FHIAIMS: https://aimsclub.fhi-berlin.mpg.de/
+.. _FHIAIMS format: https://doi.org/10.6084/m9.figshare.12413477.v1
+
+"""
+from __future__ import absolute_import
+
+import re
+from six.moves import range, zip
+from six import raise_from
+
+import itertools
+import warnings
+
+import numpy as np
+
+from . import base
+from .core import triclinic_box, triclinic_vectors
+from ..exceptions import NoDataError
+from ..lib import util
+from ..lib import mdamath
+
+
+class Timestep(base.Timestep):
+
+    def _init_unitcell(self):
+        return np.zeros((3, 3), dtype=np.float32)
+
+    @property
+    def dimensions(self):
+        """unitcell dimensions (A, B, C, alpha, beta, gamma)
+        """
+        return triclinic_box(self._unitcell[0], self._unitcell[1], self._unitcell[2])
+
+    @dimensions.setter
+    def dimensions(self, new):
+        self._unitcell[:] = triclinic_vectors(new)
+
+
+class FHIAIMSReader(base.SingleFrameReaderBase):
+    """Reader for the FHIAIMS geometry format.
+
+    .. note::
+       Geometry files are single frame-only
+
+    """
+    format = ['IN', 'FHIAIMS']
+    units = {'time': None, 'length': 'Angstrom', 'velocity': None}
+    _Timestep = Timestep
+
+    def _read_first_frame(self):
+        with util.openany(self.filename, 'rt') as fhiaimsfile:
+            relative, positions, velocities, lattice_vectors = [], [], [], []
+            skip_tags = ["#","initial_moment"]
+            oldline=''
+            for line in fhiaimsfile:
+                line = line.strip()    # perhaps not needed
+                if any([line.startswith(tag) for tag in skip_tags] ):
+                    oldline=line
+                    continue
+                if line.startswith("atom"):
+                    positions.append(
+                        list(np.asarray(line.split()[1:-1], np.float32)))
```

Comment:
Can't you do the conversion at the very end, i.e.,
```suggestion
                        line.split()[1:-1])
```
and then later
```python
positions = np.array(positions, dtype=np.float32)
```
Should improve performance because you don't go back and forth between arrays and lists and numpy can cast everything in one go.

Reply:
You're already doing 
```python
positions = np.asarray(positions, dtype=np.float32)
``` 
so definitely don't use np.array here.

Reply:
done